flutter_app/lib/mavlink_communication.dart

```diff
@@ -0,0 +1,150 @@
+import 'dart:math';
+import 'dart:typed_data';
+import 'dart:io';
+import 'dart:async';
+import 'package:flutter_libserialport/flutter_libserialport.dart';
+import 'package:dart_mavlink/mavlink.dart';
+import 'package:dart_mavlink/dialects/common.dart';
+
+enum MavlinkCommunicationConnection { tcp, serial }
+
+/// Base class for sending and receiving Mavlink messages
+class MavlinkCommunication {
+  final MavlinkParser _parser;
+
+  final StreamController<double> _yawStreamController =
+      StreamController<double>();
+  final StreamController<double> _pitchStreamController =
+      StreamController<double>();
+  final StreamController<double> _rollStreamController =
+      StreamController<double>();
+  final StreamController<double> _rollSpeedController =
+      StreamController<double>();
+  final StreamController<double> _pitchSpeedController =
+      StreamController<double>();
+  final StreamController<double> _yawSpeedController =
+      StreamController<double>();
+  final StreamController<int> _timeBootMsPitchController =
+      StreamController<int>();
+  final MavlinkCommunicationConnection _connectionType;
+
+  late Stream<Uint8List> _stream;
+  late SerialPort _serialPort;
+
+  late int _tcpPort;
+  late Socket _tcpSocket;
+
+  // Class constructor
+  MavlinkCommunication(MavlinkCommunicationConnection connectionType,
+      String connectionAddress, int port)
+      : _parser = MavlinkParser(MavlinkDialectCommon()),
+        _connectionType = connectionType {
+    switch (_connectionType) {
+      case MavlinkCommunicationConnection.tcp:
+        {
+          _tcpPort = port;
+          _startupTcpPort(connectionAddress);
+          break;
+        }
+      case MavlinkCommunicationConnection.serial:
+        {
+          _startupSerialPort(connectionAddress);
+          break;
+        }
+    }
+    _parseMavlinkMessage();
+  }
+
+  _startupTcpPort(String connectionAddress) async {
+    // Connect to the socket
+    _tcpSocket = await Socket.connect(connectionAddress, _tcpPort);
```

Comment:
Instead of a `_tcpPort` member, change `_startupTcpPort()` to accept a port parameter instead.